src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/BuildUpToDateCheck.cs

```diff
@@ -143,12 +144,13 @@ void AddImports(IReadOnlyList<IProjectImportSnapshot> value)
 
         private void OnSourceItemChanged(IProjectSubscriptionUpdate e)
         {
-            foreach (var itemType in e.ProjectChanges)
+            foreach (var itemType in e.ProjectChanges.Where(changes => changes.Value.Difference.AnyChanges))
```

Comment:
You can glob any item. `Reference` is probably rare, but the default SDK template [includes](https://github.com/dotnet/sdk/blob/feca75bd43f6374715621024d930164de8bde67c/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.DefaultItems.props#L25-L33) `Compile`, `EmbeddedResource`, and `None`, so you should try to catch those. The Web SDK [also adds](https://github.com/aspnet/websdk/blob/e50ec659a80f19b54837023c2493b73a48911373/src/ProjectSystem/Microsoft.NET.Sdk.Web.ProjectSystem.Targets/netstandard1.0/Microsoft.NET.Sdk.Web.ProjectSystem.props#L26-L44) `Content`.